packages/.gitignore

```diff
@@ -0,0 +1,3 @@
+dist
```

Comment:
(putting this comment here because it's the first file under the new /packages directory)

nit: should we make this packages/javascript/ so that we can eventually have packages for other languages? It also seems weird to call it `packages` when it only corresponds to one npm package, but if packages were set up to (eventually) contain other languages it would make more sense as a directory name.

Reply:
My idea was to break jamsocket-javascript into 4 packages that would get published separately to npm, so instead of `@jamsocket/javascript/server`, `@jamsocket/javascript/client`, `@jamsocket/javascript/react`, and `@jamsocket/javascript/socketio`, we'd have `@jamsocket/server`, `@jamsocket/client`, `@jamsocket/react`, and `@jamsocket/socketio`. I guess I could put this stuff in a `javascript` (or `typescript`?) directory for now, though, since I don't know when I can get around to breaking this into separate packages.

Reply:
a bit worried if this creates a worse dev experience in having to import four packages if you want to use all parts of the lib?

Reply:
I think we'd probably just re-export pieces of the "parent" libraries. E.g., the socketio library would itself require the client and react libraries and then re-export the client and react APIs so the dev doesn't need to explicitly import them.

The other advantages to publishing separate npm packages is that if you just wanted to use the server functions, your `npm install` wouldn't need to install the socketio client library - which is only used for the socketio piece. Also, we wouldn't have to namespace everything two levels like we do now, `@jamsocket/javascript/server` vs `@jamsocket/server`.

Reply:
got it, sounds good!